src/containers/course-section/resources-list/ResourcesList.tsx

```diff
@@ -77,47 +72,71 @@ const ResourcesList: FC<ResourcesListProps> = ({
-      {(provided: DraggableProvided, snapshot: DraggableStateSnapshot) => (
-        <Box
-          ref={provided.innerRef}
-          sx={styles.section(snapshot.isDragging)}
-          {...provided.draggableProps}
-        >
-          <Box
-            className='dragIcon'
-            sx={styles.dragIcon}
-            {...provided.dragHandleProps}
-          >
-            <DragIndicatorIcon />
-          </Box>
-          <ResourceItem
-            deleteResource={deleteResource}
-            editResource={editResource}
-            resource={item}
-            setResourceAvailability={setResourceAvailability}
-          />
-        </Box>
-      )}
-    </Draggable>
-  ))
+  const handleDragEnd = useCallback(
+    (event: DragEndEvent) => {
+      const { active, over } = event
+      if (active && over) {
+        setResources((prevResources) => {
+          const activeIndex = prevResources.findIndex(
+            (item) => item._id === active.id
+          )
+          const overIndex = prevResources.findIndex(
+            (item) => item._id === over.id
+          )
+          if (activeIndex !== -1 && overIndex !== -1) {
+            return arrayMove(prevResources, activeIndex, overIndex)
+          }
+          return prevResources
+        })
+      }
+    },
+    [setResources]
+  )
+
+  const renderItem = (item: CourseResources, isDragOver = false) => (
+    <SortableWrapper
+      id={item._id}
+      key={item._id}
+      onDragEndStyles={styles.section(isDragOver)}
+      onDragStartStyles={styles.section(true)}
+    >
+      <DragHandle iconStyles={styles.dragIcon} />
+      <ResourceItem
+        deleteResource={deleteResource}
+        editResource={editResource}
+        resource={item}
+        setResourceAvailability={setResourceAvailability}
+      />
+    </SortableWrapper>
+  )
+
+  const resourceItems = items.map((item) => renderItem(item))
 
   return (
-    <Box sx={styles.root}>
-      <DragDropContext onDragEnd={onDragEnd}>
-        {enabled && (
-          <Droppable droppableId='draggable'>
-            {(provided: DroppableProvided) => (
-              <Box {...provided.droppableProps} ref={provided.innerRef}>
-                {resourcesList}
-                {provided.placeholder}
-              </Box>
-            )}
-          </Droppable>
-        )}
-      </DragDropContext>
-    </Box>
+    <DndContext
+      onDragCancel={() => {
+        setActive(null)
+      }}
+      onDragEnd={handleDragEnd}
+      onDragStart={({ active }) => {
+        setActive(active)
+      }}
+      sensors={sensors}
+    >
+      {enabled && (
+        <>
+          <SortableContext
+            items={items.map((item) => item._id)}
+            strategy={verticalListSortingStrategy}
+          >
+            <Box sx={styles.root}>{resourceItems}</Box>
+          </SortableContext>
+          <DragOverlay>
+            {activeItem && renderItem(activeItem, true)}
+          </DragOverlay>
+        </>
+      )}
```

Comment:
Please move it outside `return` statement to avoid polluting jsx

Reply:
thank you for notice, fixed